### DISCUSSION TITLE
Svelte createMutation mutates before mutate( )

### DISCUSSION THREAD

**Discussion by danilockthar** [TanStack/query#5055] [Q&A]
Hi! im following this example in the docs but its beheaving a little bit weird in my app. 
https://codesandbox.io/s/github/tanstack/query/tree/main/examples/svelte/playground?from-embed=&file=/src/routes/AddTodo.svelte

right now im adding in my script the createMutation method  

`	const addMutation = createMutation({
    mutationFn: unsubscribe,
    onSuccess: (data) => {
		console.log(data)
    },
  })`
  
  with some `<button on:click={$addMutation.mutate()}`
  
  but in the first render the mutates happens automatically, without clicking the button
  
someone can give me a hand ? thank you

**Comment by louis-young** [comment]
Hi 👋 

I can't reproduce the issue using the sandbox you provided - are you still seeing the issue?

> with some `<button on:click={$addMutation.mutate()}`

This might explain the behaviour you're seeing. I'm primarily a React user but you likely want to pass a reference to a function that calls `mutate` instead of calling it like so:

```diff
- <button on:click={$addMutation.mutate()}
+ <button on:click={() => $addMutation.mutate({ name })}
```

Let me know if you still need a hand 🙂

**Reply by danilockthar** [reply]
Hi @louis-young  thank you for your answer!

Sorry for the lacking reproduce sb , i'l try to do one

i have tried that but its the same, without the name parameter, my function doesnt have that like the example in the docs
